device/transport/mqtt/test/_mqtt_test.js

```diff
@@ -555,9 +558,21 @@ describe('Mqtt', function () {
         testCallback();
       });
     });
+
+    it('sets options for productInfo', function(testCallback) {
+      var mqtt = new Mqtt(fakeX509AuthenticationProvider);
+      mqtt.setOptions(fakeProductInfoOptions, function (err) {
+        assert.strictEqual(mqtt._productInfo, fakeProductInfoOptions.productInfo);
+        testCallback();
```

Comment:
same as in HTTP - a better test would be to verify the username property on mqttbase.connect